src/app/core/auth-guard-with-forced-login.service.ts

```diff
@@ -7,21 +7,22 @@ import { AuthService } from './auth.service';
 
 @Injectable()
 export class AuthGuardWithForcedLogin implements CanActivate {
-  private isAuthenticated: boolean;
+
 
   constructor(
     private authService: AuthService,
   ) {
-    this.authService.isAuthenticated$.subscribe(i => this.isAuthenticated = i);
+
   }
 
   canActivate(
     route: ActivatedRouteSnapshot,
     state: RouterStateSnapshot,
   ): Observable<boolean> {
     return this.authService.isDoneLoading$
+      .pipe(withLatestFrom(this.authService.isAuthenticated$))
       .pipe(filter(isDone => isDone))
-      .pipe(tap(_ => this.isAuthenticated || this.authService.login(state.url)))
-      .pipe(map(_ => this.isAuthenticated));
+      .pipe(tap([_, isAuthenticated] => isAuthenticated || this.authService.login(state.url)))
+      .pipe(map([_, isAuthenticated] => isAuthenticated));
```

Comment:
I think that the array needs wrapping in parens, i.e. `tap(([_, isAuth]) => ....`